File: www/JSCore/Polyfills/Workers/SharedWatchWorker.js
importScripts("SharedWorker");

class SharedWatchWorker extends SharedWorker {

    constructor(scriptURL, name, instance) {
        super();
        if (this.constructor.name !== "SharedWatchWorker") { return; }
        this.instance = instance || scope && scope.createSharedWatchWorker(scriptURL, name);
    }

    static create(instance) {
        return new SharedWatchWorker(instance.scriptURL, instance.name, instance);
    }

}

File: src/ios/JSCore/Workers/SharedWatchWorker.swift
//
//  SharedWatchWorker.swift
//  WTVJavaScriptCore
//
//  Created by Jing KE on 30/07/16.
//  Copyright © 2016 WizTiVi. All rights reserved.
//

import Foundation

class SharedWatchWorker: SharedWorker{
    
    override func run() {
        print("==================== [\(self.className)] Start running shared watch worker ====================")
        SharedWatchWorkerGlobalScope.runWorker(self)
        print("==================== [\(self.className)] End running shared watch worker ====================")
    }
    
}

extension SharedWatchWorker {
    
    override class func create(context: ScriptContext, scriptURL: String, name: String = "") -> SharedWatchWorker {
        return SharedWatchWorker(context: context, scriptURL: scriptURL, name: name)
    }
    
}